### PR TITLE
fix(ticket): percentage of resolution time

### PR DESCRIPTION
**Pull request by Rom1-B**
Percentage of resolution time does not exclude waiting time. I think that the pending should freeze this evolution, so its value should be similar just before the pending and just after the exit of pending.

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !30259

### CODE REVIEW

**Comment by Rom1-B** [comment]
As I don't have the rights to modify your PR, I've recreated another one https://github.com/glpi-project/glpi/pull/15948

File: src/Search.php [hunk 1]
```diff
@@ -7122,7 +7122,8 @@ public static function giveItem(
                             }
                         }
                         if ($totaltime != 0) {
-                            $percentage  = round((100 * $currenttime) / $totaltime);
+                            $waitingtime = $item->fields['sla_waiting_duration'];
```
Comment (cedric-anne) [inline src/Search.php:7125]: The `$waitingtime` value should be 0 if there is no SLA defined. Also, the `sla_waiting_duration` is only related to the TTR SLA and should not have any impact on TTO and/or OLA. You should define this variable in the previous if/else blocks.
```php
if ($item->isField($slaField) && $item->fields[$slaField] != 0) { // Have SLA
    ...
    $waitingtime = $slaField === 'slas_id_ttr' ? $item->fields['sla_waiting_duration'] : 0;
    ... 
} else {
    ...
    $waitingtime = 0;
    ...
}
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by cedric-anne — APPROVED** [review]
Seems OK, but I would like to have the customer feedback.

**Comment by Rom1-B** [comment]
> Seems OK, but I would like to have the customer feedback.

The customer has validated

**Review by trasher — APPROVED** [review]

**Pull request merged**